### DISCUSSION TITLE
How do I make Pulsar remember what project tabs I had opened during my last session?

### DISCUSSION THREAD

**Discussion by tetron432** [pulsar-edit/.github#166] [Pulsar Support]
I remember when I was using Atom, it was the only annoying thing that nobody had a real solution to. Sometimes when I opened the app, I had to reopen all the project folders that I was previously working on, which was very inconvenient. Does Pulsar have a solution to this?

**Comment by confused-Techie** [comment]
At least for me it almost always does. Like if I open Pulsar in a directory, where it was open previously, my tabs open up in the same locations that they were in last time I worked in that directory. So if this is what you mean I'd assume they always would.

When it doesn't open up like you'd expect, how are you opening it? Are there any error messages?

**Comment by Daeraxa** [comment]
Mine always open too. You can find what ones should be opening in `~/.pulsar/storage/application.json` where it will list all the project roots it has stored.

I know it can get weird if you delete a project folder without removing it from the tree-view then try to add new projects, it seems like it struggles to update if it has a missing reference on it. In that case it should shout at you with a warning popup saying it can't reopen it.
